vggish_audio_encoder.py

```diff
@@ -0,0 +1,97 @@
+__copyright__ = "Copyright (c) 2021 Jina AI Limited. All rights reserved."
+__license__ = "Apache-2.0"
+
+import os
+from typing import Any, Optional, List, Iterable
+import tensorflow as tf
+tf.compat.v1.disable_eager_execution()
+
+from jina import Executor, requests, Document, DocumentArray
+from vggish.vggish_postprocess import *
+from vggish.vggish_slim import *
+
+cur_dir = os.path.dirname(os.path.abspath(__file__))
+
+
+def _batch_generator(data: List[Any], batch_size: int):
+    for i in range(0, len(data), batch_size):
+        yield data[i: i + batch_size]
+
+
+class VggishAudioEncoder(Executor):
+    """
+     Encode audio data with Vggish embeddings
+
+     :param model_path: path of the pre-trained Vggish model
+     :param pca_path: path of the pre-trained PCA model
+     :param default_batch_size: fallback traversal path in case there is not traversal path sent in the request
+     :param default_traversal_paths: fallback batch size in case there is not batch size sent in the request
+     """
+
+    def __init__(self,
+                 model_path: str = os.path.join(cur_dir, 'models/vggish_model.ckpt'),
+                 pca_path: str = os.path.join(cur_dir, 'models/vggish_pca_params.npz'),
+                 default_batch_size: int = 128,
+                 default_traversal_paths: Iterable[str] = ['r'],
+                 *args, **kwargs):
+
+        super().__init__(*args, **kwargs)
+        self.model_path = model_path
+        self.pca_path = pca_path
+        self.default_batch_size = default_batch_size
+        self.default_traversal_paths = default_traversal_paths
+
+        self.sess = tf.compat.v1.Session()
+        define_vggish_slim()
+        load_vggish_slim_checkpoint(self.sess, self.model_path)
+        self.feature_tensor = self.sess.graph.get_tensor_by_name(
+            INPUT_TENSOR_NAME)
+        self.embedding_tensor = self.sess.graph.get_tensor_by_name(
+            OUTPUT_TENSOR_NAME)
+        self.post_processor = Postprocessor(self.pca_path)
+
+    def post_init(self):
```

Comment:
dont forget to remove `post_init` and merge into `init` if necessary

Reply:
yes, removed.